src/form/Form.jsx

```diff
@@ -104,15 +116,18 @@ export default class Form extends React.Component {
 
   onSubmit = (event) => {
     event.preventDefault()
-
-    if (!this.props.disabled) {
+    if (!this.props.disabled && !this.state.submitting) {
       const errors = this.props.validate(this.state.value, true)
 
       if (containsError(errors)) {
         this.setState({errors, triedToSubmit: true})
       } else {
         this.setState({errors: new Immutable.Map()})
-        this.props.onSubmit(this.props.normalize(this.state.value))
+        const result = this.props.onSubmit(this.props.normalize(this.state.value))
+        if (result && typeof result.then === 'function') {
+          this.setState({submitting: true})
+          result.catch(() => {}).then(() => this.setState({submitting: false}))
```

Comment:
Why dont we have to reset submitting state after an error occurred? Didn't it stop submitting then too?